sys/include/net/gnrc/priority_pktqueue.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @defgroup    net_gnrc_priority_pktqueue Priority packet queue
+ * @ingroup     net_gnrc
+ * @file
+ * @brief       Wrapper for priority_queue that holds gnrc_pktsnip_t* and is
+ *              aware of it's length.
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#ifndef GNRC_PRIORITY_PKTQUEUE_H
+#define GNRC_PRIORITY_PKTQUEUE_H
+
+#include <stdint.h>
+#include <priority_queue.h>
+#include <net/gnrc/pkt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct gnrc_priority_pktqueue_node {
+    struct gnrc_priority_pktqueue_node *next;   /**< next queue node */
+    uint32_t priority;                          /**< queue node priority */
+    gnrc_pktsnip_t *pkt;                        /**< queue node data */
+} gnrc_priority_pktqueue_node_t;
+
+/* TODO: Description */
+typedef priority_queue_t gnrc_priority_pktqueue_t;
+
+#define PRIORITY_PKTQUEUE_NODE_INIT(priority, pkt) { NULL, priority, pkt }
+
+static inline void priority_pktqueue_node_init(gnrc_priority_pktqueue_node_t *node,
```

Comment:
Please prefix these functions and macros with `gnrc_`. I know I introduced some of these, but I tried to keep style with the current status of this PR.